lib/components/term.js

```diff
@@ -91,6 +91,23 @@ export default class Term extends React.PureComponent {
   componentDidMount() {
     const {props} = this;
 
+    if (props.term) {
+      let _core = props.term._core;
+      // Release listeners and disposables
+      _core._disposables.forEach(d => d.dispose());
+      _core._disposables.length = 0;
+      _core._events = {};
```

Comment:
Not all listeners are installed when open is called, doing this on multiple objects will almost certainly break other components. Can the code be changed to call `open` once? I think it works fine if you open on some parent element and transfer that between split elements.

Also here is upstream request https://github.com/xtermjs/xterm.js/issues/1323

Reply:
@Tyriar Thanks! The only listener not registered in `open` that I noticed was InputHandler. In the end the object ends up with the same number of listeners as before the `dispose` calls. But yes I agree that calling `open` once and reattaching the container as you suggest would be better, not sure how much more complicated it is with React but it should be possible.